src/classes/ImportBackendKeywordsNew.cls

```diff
@@ -0,0 +1,158 @@
+global class ImportBackendKeywordsNew {
+  /*
+    Takes in adapter which is a "linked" list (by path) and menu Label which should be 'CKW Search 3.0'
+    returns true if successful, false otherwise
+   */
+  webservice static boolean importBackendKeywords(List<MenuItemAdapter> menuItemAdapters, String menuLabel) {
+    if (menuItemAdapters == null || menuItemAdapters.size() == 0) {
+      return false;
+    }
+    Menu__c menu = getMenuByLabel(menuLabel);
+    List<Menu_Item__c> updateMenuItems = new List<Menu_Item__c>();
+    List<Menu_Item__c> insertMenuItems = new List<Menu_Item__c>();
+    for (MenuItemAdapter currentAdapter : menuItemAdapters) {
+      // Skip adapter if it has already been processed.
+      if (currentAdapter.IsProcessed) {
+        continue;
+      }
+      Menu_Item__c foundItem = getRelatedMenuItem(currentAdapter, menuItemAdapters, menu.Id);
+      // check if the menu item has been found if yes update, if no, create menu item, add it to list then continue
+      if (null != foundItem) {
+        currentAdapter.MenuItem = updateMenuItemContent(currentAdapter, foundItem);
+        // Work around so as to not add duplicates in the list
+        // Better alternative compared to iterating over list each time
+        // which would exponetially increase script statements
+        try {
+          updateMenuItems.add(currentAdapter.MenuItem);
+        } catch (ListException e) {
+          // Empty catch intentional!
+        }
+        continue;
+      }
+      currentAdapter.MenuItem = createAndSaveMenuItem(currentAdapter, menu, menuItemAdapters);
+      insertMenuItems.add(currentAdapter.MenuItem);
+    }
+    System.debug(Logginglevel.DEBUG, 'Count of menu items to update is ' + updateMenuItems.size());
+    System.debug(Logginglevel.DEBUG, 'Count of menu items to insert is ' + insertMenuItems.size());
+    if (updateMenuItems.size() > 0) {
+      // update List of items to be updated
+      database.update(updateMenuItems);
+    }
+    return true;
+  }
+
+  // get menu item by
+  public static Menu__c getMenuByLabel(String label) {
+    return [SELECT
+            Id, Label__c
+            FROM
+            Menu__c
+            WHERE
+            Label__c = :label
+            LIMIT 1];
+  }
+
+  // check if sent menu item exists in salesforce, returns it if it exists
+  private static Menu_Item__c getRelatedMenuItem(MenuItemAdapter menuItemAdapter, List<MenuItemAdapter> adapters, String menuId) {
+    MenuItemAdapter previousAdapter = findAdapterByPath(menuItemAdapter.PreviousItemPath, adapters);
+    List<Menu_Item__c> menuItems = null;
+
+    if (previousAdapter != null && previousAdapter.MenuItem != null) {
+      menuItems = [
+                    SELECT
+                    Id, Label__c, Attribution__c, Content__c, Is_Active__c, Parent_Item__c, Last_Modified_Date__c
```

Comment:
You need to resolve your syntax. You can ask to be guided on this matter so that it doesnt have to persist in your next commits